compiler/semmagic.nim

```diff
@@ -49,7 +49,9 @@ proc semTypeOf(c: PContext; n: PNode): PNode =
     else:
       m = mode.intVal
   result = newNodeI(nkTypeOfExpr, n.info)
+  inc c.inTypeofContext
   let typExpr = semExprWithType(c, n[1], if m == 1: {efInTypeof} else: {})
```

Comment:
As you can see here we already have `efInTypeof`. Can this be used instead of `c.inTypeofContext`?

Reply:
That's what I tried first, it doesn't always propagate, even in simple cases like `foo(compileTimeProc())` where it's nested in another call.